interview-150/151.ts

```diff
@@ -0,0 +1,10 @@
+function reverseWords(s: string): string {
+    // remove whitespace from input string using regexp and trim
+    s = s.replace(/ +(?= )/g,'').trim()
+
+    // convert the string to an array of strings using split
+    const arr: string[] = s.split(" ")
+
+    // reverse the array and return it as a string with reverse and join
+    return arr.reverse().join(" ")
```

Comment:
These comments are close to what I'm looking for! Maybe a _little_ too obvious though - I'd do more like:
```ts
// remove all duplicate/excess white space from input
s = s.replace(/ +(?= )/g,'').trim()

// create an array of words
const words: string[] = s.split(" ")

// reverse & join again, adding the whitespace back.
return arr.reverse().join(" ")
```